File: Problems/0198_House_Robber/House_Robber.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int Rob(int[] nums)
    {
        if (nums.Length == 0)
            return 0;
        if (nums.Length == 1)
            return nums[0];

        int[] c = new int[nums.Length];
        int max;

        c[0] = nums[0];
        if (nums[0] >= nums[1])
            c[1] = nums[0];
        else
            c[1] = nums[1];

        max = c[1];
        
        for (int i = 2; i < nums.Length; ++i) {
            if (nums[i] + c[i - 2] >= c[i - 1])
                c[i] = nums[i] + c[i - 2];
            else
                c[i] = c[i - 1];
            
            if (max < c[i])
                max = c[i];
        }

        return max;
    }

    private int[] array_str_to_int(string[] workStr)
    {
        int[] numbers = new int[workStr.Length];

        for (int i = 0; i < workStr.Length; i++) {
            numbers[i] = int.Parse(workStr[i]);
        }

        return numbers;
    }

    private string output_array_int(int[] numbers)
    {
        if (numbers.Length == 0)
            return "";

        string resultStr = numbers[0].ToString();
        for (int i = 1; i < numbers.Length; i++)
            resultStr += "," + numbers[i].ToString();

        return resultStr;
    }

    public void Main(string args)
    {
        string[] numbersStr = args.Replace("[","").Replace("]","").Split(',');
        int[] nums = array_str_to_int(numbersStr);

        Console.WriteLine("numbers = [" + output_array_int(nums) + "]");

        System.Diagnostics.Stopwatch sw = new System.Diagnostics.Stopwatch();
        sw.Start();
        
        int result = Rob(nums);

        Console.WriteLine("result = " + result);
        
        sw.Stop();
        Console.WriteLine("Execute time ... " + sw.ElapsedMilliseconds.ToString() + "ms");
    }
}

File: Problems/0100_0199/0198_House_Robber/Project_Python3/House_Robber.py
# coding: utf-8

import os
import sys
import time

class Solution:
    def rob(self, nums):
        if len(nums) == 0:
            return 0
        if len(nums) == 1:
            return nums[0]
        c = [0] * len(nums)
        c[0] = nums[0]
        c[1] = max(nums[0] ,nums[1])
        for i in range(2, len(nums)):
            c[i] = max(nums[i] + c[i-2], c[i-1])
        return max(c)

def array_str_to_int(numbersStr):
    numbers = [0]*len(numbersStr)
    for i in range(len(numbersStr)):
        numbers[i] = int(numbersStr[i])
    return numbers

def main():
    argv = sys.argv
    argc = len(argv)

    if argc < 2:
        print("Usage: python {0} <testdata.txt>".format(argv[0]))
        exit(0)

    if not os.path.exists(argv[1]):
        print("{0} not found...".format(argv[1]))
        exit(0)

    testDataFile = open(argv[1], "r")
    lines = testDataFile.readlines()

    for temp in lines:
        temp = temp.strip()
        if temp == "":
            continue
        print("args = {0}".format(temp))
        loop_main(temp)
    #   print("Hit Return to continue...")
    #   input()

def loop_main(temp):
    numsStr = temp.replace("[","").replace("]","").split(",")
    nums = array_str_to_int(numsStr)

    sl = Solution()
    time0 = time.time()
    result = sl.rob(nums)
    print("result = {0}".format(result))

    time1 = time.time()

    print("Execute time ... : {0:f}[s]\n".format(time1 - time0))

if __name__ == "__main__":
    main()
